### PR TITLE
Bump github.com/gardener/gardener to 1.85.1

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area open-source
/kind enhancement
/platform aws

**What this PR does / why we need it**:
This PR updates the `github.com/gardener/gardener` dependency to `v1.85.1`. As a consequence, admission CA and service certificates are now managed automatically. The vendor directory is removed as well.

**Release note**:

```other operator
Bump github.com/gardener/gardener to 1.85.1.
```
```breaking operator
CA and server certificates for the admission component are managed automatically. Passing custom certificates via Helm values is not supported anymore.
```
```other developer
The `vendor` directory was removed in favor of the `go mod cache`.
```

### CODE REVIEW

**Comment by timuthy** [comment]
Vendor dir removal happened with the help of https://github.com/gardener/gardener-extension-provider-openstack/pull/698. However, the K8s code generation is broken now. I asked @afritzler and @MartinWeindel to double check the adjustments (mainly https://github.com/gardener/gardener-extension-provider-aws/pull/844/commits/a3b7567403255d9ad7a83f235e9915217f8222a1).

File: Dockerfile [hunk 1]
```diff
@@ -2,8 +2,13 @@
 FROM golang:1.21.1 AS builder
 
 WORKDIR /go/src/github.com/gardener/gardener-extension-provider-aws
+
 COPY . .
 
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
```
Comment (MartinWeindel) [inline Dockerfile:11]: Caching of deps should happen before copying source code.
```suggestion

# Copy go mod and sum files
COPY go.mod go.sum ./
# Download all dependencies. Dependencies will be cached if the go.mod and go.sum files are not changed
RUN go mod download

COPY . .

```

File: hack/update-codegen.sh [hunk 2]
```diff
@@ -18,6 +18,11 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+CODE_GEN_DIR=$(go list -m -f '{{.Dir}}' k8s.io/code-generator)
+
+# setup virtual GOPATH
+source "$GARDENER_HACK_DIR"/vgopath-setup.sh
```
Comment (MartinWeindel) [inline hack/update-codegen.sh:24]:
```suggestion
CODE_GEN_DIR=$(go list -m -f '{{.Dir}}' k8s.io/code-generator)

# setup virtual GOPATH
source "$GARDENER_HACK_DIR"/vgopath-setup.sh
```

**Review by MartinWeindel — CHANGES_REQUESTED** [review]
some minor improvements

**Review by MartinWeindel — APPROVED** [review]
/lgtm

**Pull request merged**